examples/table_script/table_script.py

```diff
@@ -0,0 +1,157 @@
+                              },
+                      'R2': {'train': [],
+                               'val': [],
+                               'test': [],
+                            },
+                     }
+        
+        # obtain indices
+        _,_,_, train_indices, val_indices, test_indices = train_val_test_split(X,
+                                                                        train_size=0.85,
+                                                                        val_size=0.05,
+                                                                        test_size=0.1,
+                                                                        sampler=sampler,
+                                                                        random_state=seed,
+                                                                        hopts=hopts,
+                                                                        return_indices=True,
+                                                                        ) 
+        
+        
+        # create data splits
+        X_train = X[train_indices]
+        X_val = X[val_indices]
+        X_test = X[test_indices]
+        
+        y_train = y[train_indices]
+        y_val = y[val_indices]
+        y_test = y[test_indices]
+        
+        
+        # fit the model to the training data
+        sklearn_model.fit(X_train, y_train)
+        
+        # get predictions
+        y_pred_train = sklearn_model.predict(X_train)
+        y_pred_val = sklearn_model.predict(X_val)
+        y_pred_test = sklearn_model.predict(X_test)
+        
+        
+        # store MAEs
+        train_mae = mean_absolute_error(y_train, y_pred_train)
+        error_dict['mae']['train'].append(train_mae)
+
+        val_mae = mean_absolute_error(y_val, y_pred_val)
+        error_dict['mae']['val'].append(val_mae)
+
+        test_mae = mean_absolute_error(y_test, y_pred_test)
+        error_dict['mae']['test'].append(test_mae)
+        
+        
+        # store RMSEs
+        train_rmse = mean_squared_error(y_train, y_pred_train, squared=False)
+        error_dict['rmse']['train'].append(train_rmse)
+
+        val_rmse = mean_squared_error(y_val, y_pred_val, squared=False)
+        error_dict['rmse']['val'].append(val_rmse)
+
+        test_rmse = mean_squared_error(y_test, y_pred_test, squared=False)
+        error_dict['rmse']['test'].append(test_rmse)
+
+        
+        # store R2
+        train_R2 = r2_score(y_train, y_pred_train)
+        error_dict['R2']['train'].append(train_R2)
+
+        val_R2 = r2_score(y_val, y_pred_val)
+        error_dict['R2']['val'].append(val_R2)
+
+        test_R2 = r2_score(y_test, y_pred_test)
+        error_dict['R2']['test'].append(test_R2)
+        
+        final_dict[sampler] = error_dict
+        
+    return final_dict
+
+
+# use default hyperparameters
+sklearn_model = LinearSVR()
+
+final_dict = sample_function(sklearn_model, X, y)
+pprint(final_dict)
```

Comment:
I am partial to [`tabulate`](https://github.com/astanin/python-tabulate) for printing out nicely formatted tables

Reply:
Oops I forgot you had already included this as a dependency. Tabulate does seem to have nicer formatting. However, a quick glance through their GitHub seems like it doesn't play nicely with nested dictionaries? I'm happy to learn more if you have more experience though. Potentially, the main deliverable is just returning the dictionary though and letting the user format as desired?

Reply:
The output is an interesting question - I think the dictionary is a really good idea, since that will play much nicer with any follow up plotting or exporting they want to do. The table I was imagining was only for short-running applications, quick 'scans' of which sampler might be the most interesting/challenging/informative/etc.